receiver/vcenterreceiver/metrics.go

```diff
@@ -0,0 +1,245 @@
+		// right now we are only grabbing real time metrics from the performance
+		// manager
+		IntervalId: int32(20),
+	}
+
+	info, err := v.client.performanceQuery(ctx, spec, hostPerfMetricList, []types.ManagedObjectReference{host.Reference()})
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+	v.processHostPerformance(info.results)
+}
+
+// vmPerfMetricList may be customizable in the future but here is the full list of Virtual Machine Performance Counters
+// https://docs.vmware.com/en/vRealize-Operations/8.6/com.vmware.vcom.metrics.doc/GUID-1322F5A4-DA1D-481F-BBEA-99B228E96AF2.html
+var vmPerfMetricList = []string{
+	// network metrics
+	"net.packetsTx.summation",
+	"net.packetsRx.summation",
+	"net.bytesRx.average",
+	"net.bytesTx.average",
+	"net.usage.average",
+
+	// disk metrics
+	"disk.write.average",
+	"disk.totalWriteLatency.average",
+	"virtualDisk.totalWriteLatency.average",
+}
+
+func (v *vcenterMetricScraper) recordVMPerformance(
+	ctx context.Context,
+	vm mo.VirtualMachine,
+	errs *scrapererror.ScrapeErrors,
+) {
+	spec := types.PerfQuerySpec{
+		Entity: vm.Reference(),
+		Format: string(types.PerfFormatNormal),
+		// Just grabbing real time performance metrics of the current
+		// supported metrics by this receiver. If more are added we may need
+		// a system of making this user customizable or adapt to use a 5 minute interval per metric
+		IntervalId: int32(20),
+	}
+
+	info, err := v.client.performanceQuery(ctx, spec, vmPerfMetricList, []types.ManagedObjectReference{vm.Reference()})
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	v.processVMPerformanceMetrics(info)
+}
+
+func (v *vcenterMetricScraper) processVMPerformanceMetrics(info *perfSampleResult) {
+	for _, m := range info.results {
+		for _, val := range m.Value {
+			for j, nestedValue := range val.Value {
+				si := m.SampleInfo[j]
+				switch val.Name {
+				// Performance monitoring level 1 metrics
+				case "net.bytesTx.average":
+					v.mb.RecordVcenterVMNetworkThroughputDataPoint(pdata.NewTimestampFromTime(si.Timestamp), nestedValue, metadata.AttributeThroughputDirectionTransmitted)
+				case "net.bytesRx.average":
+					v.mb.RecordVcenterVMNetworkThroughputDataPoint(pdata.NewTimestampFromTime(si.Timestamp), nestedValue, metadata.AttributeThroughputDirectionReceived)
+				case "net.usage.average":
+					v.mb.RecordVcenterVMNetworkUsageDataPoint(pdata.NewTimestampFromTime(si.Timestamp), nestedValue)
+				case "net.packetsTx.summation":
+					v.mb.RecordVcenterVMNetworkPacketCountDataPoint(pdata.NewTimestampFromTime(si.Timestamp), nestedValue, metadata.AttributeThroughputDirectionTransmitted)
+				case "net.packetsRx.summation":
+					v.mb.RecordVcenterVMNetworkPacketCountDataPoint(pdata.NewTimestampFromTime(si.Timestamp), nestedValue, metadata.AttributeThroughputDirectionReceived)
+
+				// Performance monitoring level 2 metrics required
+				case "disk.totalReadLatency.average", "virtualDisk.totalReadLatency.average":
```

Comment:
Usually VMs will either be virtually managed or physically managed. Because of this I assumed/think it will be safe not to overwrite. 

This document outlines a little more information on the differences between the two types of disks VMs can utilize: 

https://www.liquidweb.com/kb/vmware-disk-types/

Reply:
Seems like something worth knowing for sure. Can you let me know what you figure out? Getting this wrong would lead to non-deterministic behavior that would likely be quite difficult to troubleshoot.

I also noticed that this metric and quite a few others are not included in the integration or unit level tests. It's not strictly necessary that all of them are, but it would be good to double check. This metric is particularly worrisome if we don't have at least one example showing it works as expected.